### PR TITLE
feat: add generic border API (Android)

### PR DESCRIPTION
**Pull request by m1ga**
Android parity for https://github.com/appcelerator/titanium_mobile/pull/11113

**Optional Description:**

```javascript
var win = Ti.UI.createWindow({
	backgroundColor: '#333',
	layout: "horizontal"
});

var view1 = Ti.UI.createView({
	backgroundColor: '#fff',
	height: 200,
	borderRadius: 20,
	width: 200,
	left: 10,
	top: 10,
	borderRadiusEdges: [Ti.UI.BORDER_EDGE_TOP_RIGHT, Ti.UI.BORDER_EDGE_TOP_LEFT]
});
var view2 = Ti.UI.createView({
	backgroundColor: '#fff',
	height: 200,
	borderRadius: 20,
	width: 200,
	left: 10,
	top: 10,
	borderRadiusEdges: [Ti.UI.BORDER_EDGE_TOP_RIGHT, Ti.UI.BORDER_EDGE_BOTTOM_LEFT]
});
var view3 = Ti.UI.createView({
	backgroundColor: '#fff',
	height: 200,
	borderRadius: 20,
	width: 200,
	left: 10,
	top: 10,
	borderRadiusEdges: [Ti.UI.BORDER_EDGE_TOP_LEFT, Ti.UI.BORDER_EDGE_BOTTOM_RIGHT]
});
var view4 = Ti.UI.createView({
	backgroundColor: '#fff',
	height: 200,
	borderRadius: 20,
	width: 200,
	left: 10,
	top: 10,
	borderRadiusEdges: [Ti.UI.BORDER_EDGE_BOTTOM_LEFT, Ti.UI.BORDER_EDGE_BOTTOM_RIGHT]
});
var view5 = Ti.UI.createView({
	backgroundColor: '#fff',
	height: 200,
	borderRadius: 100,
	width: 200,
	left: 10,
	top: 10,
	borderRadiusEdges: [Ti.UI.BORDER_EDGE_TOP_LEFT, Ti.UI.BORDER_EDGE_TOP_RIGHT]
});
var view6 = Ti.UI.createView({
	backgroundColor: '#fff',
	height: 200,
	borderRadius: 20,
	width: 200,
	left: 10,
	top: 10
});

win.add([view1, view2, view3, view4, view5, view6]);
win.open();
```

image

### CODE REVIEW

File: android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java [hunk 1]
```diff
@@ -1523,6 +1550,29 @@ private void handleBorderProperty(String property, Object value)
...
+						case BORDER_EDGE_BOTTOM_RIGHT:
+							radii |= BORDER_EDGE_BOTTOM_RIGHT;
+							break;
+						case BORDER_EDGE_BOTTOM_LEFT:
+							radii |= BORDER_EDGE_BOTTOM_LEFT;
+							break;
+					}
+				}
+			}
+			borderView.setBorderEdges(radii);
```
Comment (hansemannn) [inline android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java]: This looks like duplicate code. Can you move it to a helper method?

File: android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java [hunk 2]
```diff
@@ -71,8 +45,31 @@
...
+import org.appcelerator.kroll.common.TiMessenger;
+import org.appcelerator.titanium.TiApplication;
+import org.appcelerator.titanium.TiC;
+import org.appcelerator.titanium.TiDimension;
+import org.appcelerator.titanium.proxy.TiViewProxy;
+import org.appcelerator.titanium.util.TiAnimationBuilder;
+import org.appcelerator.titanium.util.TiAnimationBuilder.TiMatrixAnimation;
+import org.appcelerator.titanium.util.TiConvert;
+import org.appcelerator.titanium.util.TiUIHelper;
+import org.appcelerator.titanium.view.TiCompositeLayout.LayoutParams;
```
Comment (hansemannn) [inline android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java]: Do we need to move all imports? Or is it the linting? I thought all sources are already linted by now.

Comment (m1ga) [reply android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java]: just clang linting. Looks like it wasn't in alphabetic order before. Perhaps clang didn't do that before?

File: android/titanium/src/java/org/appcelerator/titanium/view/TiBorderWrapperView.java [hunk 3]
```diff
@@ -69,14 +69,68 @@ protected void onDraw(Canvas canvas)
...
 			} else {
 				outerPath.addRect(innerRect, Direction.CW);
 			}
 			Path innerPath = new Path(outerPath);
 
 			// draw border
-			outerPath.addRoundRect(outerRect, radius, radius, Direction.CCW);
+			if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP && radii > 0) {
+				// custom edge radius
+				float[] corners = new float[] { 0, 0, 0, 0, 0, 0, 0, 0 };
```
Comment (m1ga) [inline android/titanium/src/java/org/appcelerator/titanium/view/TiBorderWrapperView.java]: I'll put this in a own function in the next update!

File: android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java (hunk 2 above)
Comment (hansemannn) [reply android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java]: Good question. Someone should run that command through all sources (cc @jquick-axway).

**Comment by hansemannn** [comment]
@m1ga: We should start maintaining an own fork. This will never reach attention here.

**Comment by m1ga** [comment]
closing in favor of #11796

**Pull request closed without merging**